File: 3. lab/3.lab.14/src/main/java/plugins/Tiger.java
package plugins;

import model.Animal;

/**
 * @author Antonio Bukovac
 */
public class Tiger extends Animal {

    private String name;

    public Tiger(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }

    @Override
    public String greet() {
        return "Bok ja sam tigar";
    }

    @Override
    public String menu() {
        return "Volim takine";
    }

}

File: 3. lab/1.1/tiger.c
#include <stdlib.h>
#include<malloc.h>

typedef char const* (*PTRFUN)();

typedef struct {
	PTRFUN* vtable;
	char const* name;
} Tiger;

char const* name(Tiger* tiger){
    return tiger->name;
}

char const* greet(void){
    return "Bok ja sam tigar!";
}

char const* menu(void){
    return "Takin";
}

PTRFUN vtable[] = {name, greet, menu};

void construct(Tiger* tiger, char const* name){
    tiger->vtable = vtable;
    tiger->name = name;
}

void* create(char const* name){
    Tiger* tiger = (Tiger*) malloc(sizeof(Tiger));
    construct(tiger, name);
    return tiger;
}

